### PR TITLE
[clap-v3-utils, keygen] Refactor mnemonics, derivation path, and argument parsing clap logic in keygen

### PR DESCRIPTION
**Pull request by samkim-crypto**
#### Problem
The cli tool `solana-zk-keygen`, which is an analogue for `solana-keygen` for ElGamal/AE keys, is in process of development. Some of the cli logic pertaining to mnemonics, derivation path, and keygen arguments are exactly the same in `keygen` and `zk-keygen`.

#### Summary of Changes
Refactor the cli logic in `keygen` related to mnemonics, derivation path, and keygen arguments into clap-v3-utils so that both `solana-keygen` and `solana-zk-keygen` can share them.

Fixes #

### CODE REVIEW

File: clap-v3-utils/src/lib.rs [hunk 1]
```diff
@@ -23,10 +23,13 @@ impl std::fmt::Debug for DisplayError {
...
 }
 
+pub mod derivation_path;
 pub mod fee_payer;
 pub mod input_parsers;
 pub mod input_validators;
+pub mod keygen;
 pub mod keypair;
 pub mod memo;
+pub mod mnemonic;
```
Comment (CriesofCarrots) [inline clap-v3-utils/src/lib.rs:33]: Wdyt about making derivation_path and mnemonic modules within keygen?

**Review by CriesofCarrots — COMMENTED** [review]
I'm happy with moving these elements. I added one organization comment.
Wouldn't hurt to get Trent's ✅ too, if you can wait until he's back, as he had some opinions about this iirc

File: clap-v3-utils/src/lib.rs (hunk 1 above)
Comment (samkim-crypto) [reply clap-v3-utils/src/lib.rs:33]: Yes, I think that is a good idea!

**Review by CriesofCarrots — APPROVED** [review]
lgtm

**Comment by samkim-crypto** [comment]
@t-nelson When you have a chance, can you take a look at these changes :pray:?

**Review by t-nelson — APPROVED** [review]
:shipit:

**Pull request merged**